pydriller/metrics/process/commit_count.py

```diff
@@ -0,0 +1,40 @@
+"""
+Module that calculates the number of commits made to a file.
+"""
+
+from pathlib import Path
+from pydriller.domain.commit import ModificationType
+from pydriller.repository_mining import RepositoryMining
+from pydriller.metrics.process.process_metric import ProcessMetric
+
+
+class CommitCount(ProcessMetric):
+    """
+    This class is responsible to implement the Commit Count metric to \
+    measure the number of commits made to a file
+    """
+
+
+    def count(self):
+        count = 0
+        filepath = self.filepath
+
+        for commit in RepositoryMining(path_to_repo=self.path_to_repo, 
+                                       from_commit=self.from_commit,
+                                       to_commit=self.to_commit,
+                                       reversed_order=True).traverse_commits():
+
+            for modified_file in commit.modifications:
+
+                if filepath in (modified_file.new_path,
+                                modified_file.old_path):
+                    count += 1
+                    if modified_file.change_type == ModificationType.RENAME:
+                        filepath = str(Path(modified_file.old_path))
+
+                    break
+
+            if self.release_scope and commit.hash in self.releases:
```

Comment:
I meant the metrics to be release-duration, as described in the paper.
```self.releases``` in the ```ProcessMetrics``` class contains all the tagged commits for a given repo.  
It is used to find the release a given commit (to_commit) belongs to and to stop the count of the metric.

Imagine we want to calculate the metric for commit c6 below (where R1 and R2 are the tagged commits c1 and c4, respectively):

|---c2---c3---|---c5---c6---
R1-----------R2

The only commits to take into account are c4, c5 and c6.
Thus, the function ```traverse_commits()``` will yield c6, c5, c4, c3, c2, c1, but the metric count will stop at c4.

If you prefer to stop at ```from_commit``` and let decide the user the temporal range for the analysis, I will change it.

Honestly, I am now realizing that lot of projects do not tag commits, and this would limit the usefulness of the tool.

Reply:
|---c2---c3---|---c5---c6
R1-----------R2

Ok. To me, these are the possible scenarios:
1. the user wants to run the analysis **between R1 and R2**: in this case, the user would have to know what commits are R1 and R2; hence, our `self.releases` is useless because the user already knows the commits
2. the user wants to run the analysis **from c6 to R2**: the user doesn't know the commit of R2, hence he relies on our `self.releases`. However, complicating the code for just this case is not worth it. The user can easily calculate the `to_commit` by himself 

hence, I suggest to just delete the release. I can do it by myself later when I merge the PR

Reply:
ok, let's go for scenario 1. I will update the code. no worries :)